core/src/main/java/org/apache/druid/timeline/DataSegment.java

```diff
@@ -146,7 +146,7 @@ public DataSegment(
     // dataSource
     this.dimensions = prepareDimensionsOrMetrics(dimensions, DIMENSIONS_INTERNER);
     this.metrics = prepareDimensionsOrMetrics(metrics, METRICS_INTERNER);
-    this.shardSpec = (shardSpec == null) ? NoneShardSpec.instance() : shardSpec;
+    this.shardSpec = (shardSpec == null) ? new NumberedShardSpec(0, 1) : shardSpec;
```

Comment:
Any reason not to have a static shared instance for `NumberedShardSpec(0, 1)` the same way we did for `NoneShardSpec`?

Reply:
Well, we can. But I think it would be not common to create a dataSegment with null shardSpec. I guess this code is just for backward compatibility to support a really old version of dataSegment. In recent versions, every dataSegment must have a proper shardSpec.

Reply:
Oh, `DataSegment.Builder` is mostly for benchmark or unit tests. It's not supposed to be used in production codes.
I don't think the builder pattern is appropriate for DataSegment because it requires for every fields to be filled except `loadSpec` in its constructor. But, it's quite prevalent in unit tests, so I left it.

Reply:
Ah, just saw a bunch of usages, didn't look too closely at what they were :+1: